### PR TITLE
`pip>23.1` compliance

### LINKED ISSUE #621: Cannot build/install t-route modules using pip >= 23.1
When attempting to build with `pip 23.1`, the follow error occurs

## Current behavior
run `compiler.sh` and get the following error:
```sh
running build_ext
Processing /Users/nelsfrazier/workspace/ngen/utilities/hydrofabric_test/t-route/src/troute-network
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [17 lines of output]
      Traceback (most recent call last):
        File "/Users/nelsfrazier/workspace/ngen/utilities/hydrofabric_test/venv/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/nelsfrazier/workspace/ngen/utilities/hydrofabric_test/venv/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/Users/nelsfrazier/workspace/ngen/utilities/hydrofabric_test/venv/lib/python3.9/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/dk/f2t2z5mx6y161tmr6yyf3vlc0000gp/T/pip-build-env-73sgqv7z/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/private/var/folders/dk/f2t2z5mx6y161tmr6yyf3vlc0000gp/T/pip-build-env-73sgqv7z/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/dk/f2t2z5mx6y161tmr6yyf3vlc0000gp/T/pip-build-env-73sgqv7z/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super(_BuildMetaLegacyBackend,
        File "/private/var/folders/dk/f2t2z5mx6y161tmr6yyf3vlc0000gp/T/pip-build-env-73sgqv7z/overlay/lib/python3.9/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
      ModuleNotFoundError: No module named 'numpy'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```

## Expected behavior

Successful build/install

## Steps to replicate behavior (include URLs)

1. `pip install -U pip==23.1`
2. `./compiler.sh`

## Screenshots

image

### PR DESCRIPTION
**Pull request by aaraney**
[`pip` `23.1`](https://pip.pypa.io/en/stable/news/#id80) changed the default behavior when the `wheel` package is not installed. `troute.network` and `troute.routing` both require `wheel` at build time and previously did not contain `pyproject.toml` files.

> When the `wheel` package is not installed, `pip` now uses the default build backend instead of `setup.py install` and `setup.py develop` for project without `pyproject.toml`. ([#8559](https://github.com/pypa/pip/issues/8559))

The PR addresses these issues and makes it possible to install `troute` using the latest `pip` version (`24.0`).

Fixes #621.

### CODE REVIEW

**Comment by aaraney** [comment]
@robertbartel we should be able to change thing in our NextGen dockerfile once this drops.

File: compiler.sh [hunk 1]
```diff
@@ -87,8 +87,8 @@ if [[ "$build_framework" == true ]]; then
   rm -rf build
   ##python setup.py --use-cython install
   ##python setup.py --use-cython develop
-  CC=${CC} python setup.py build_ext --inplace --use-cython || exit
```
Comment (hellkite500) [inline compiler.sh:90]: Not sure how often this is used in practice, but may need to account for customized c compiliers for the cython builds...This isn't too had to add back or do in different ways without this here, so I don't see a problem in dropping it right now unless something is really depending on this...

**Review by hellkite500 — APPROVED** [review]

File: compiler.sh (hunk 1 above)
Comment (aaraney) [reply compiler.sh:90]: I removed it b.c. I thought it was [redundant](https://github.com/NOAA-OWP/t-route/blob/aeb5c56ee720685a60599db3bc0a33deca73ed29/compiler.sh#L22). Maybe i'm wrong?? `CC` is always exported (at the top of the build script). I thought that child processes would inherit the environment variables of the parent? Is that not always the case?

**Comment by aaraney** [comment]
Thanks for having a look at this, @hellkite500! Much appreciated! :tada:

**Pull request merged**